src/material-experimental/mdc-dialog/dialog-container.ts

```diff
@@ -0,0 +1,176 @@
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  host: {
+    'class': 'mat-mdc-dialog-container mdc-dialog',
+    'tabindex': '-1',
+    'aria-modal': 'true',
+    '[attr.id]': '_id',
+    '[attr.role]': '_config.role',
+    '[attr.aria-labelledby]': '_config.ariaLabel ? null : _ariaLabelledBy',
+    '[attr.aria-label]': '_config.ariaLabel',
+    '[attr.aria-describedby]': '_config.ariaDescribedBy || null',
+    '[class._mat-animation-noopable]': '!_animationsEnabled',
+  },
+})
+export class MatDialogContainer extends MatDialogContainerBase implements AfterViewInit, OnDestroy {
+  /** Whether animations are enabled. */
+  _animationsEnabled: boolean = this._animationMode !== 'NoopAnimations';
+  /** Host element of the dialog container component. */
+  private _hostElement: HTMLElement = this._elementRef.nativeElement;
+  /** Duration of the dialog open animation. */
+  private _openAnimationDuration =
+      this._animationsEnabled ? numbers.DIALOG_ANIMATION_OPEN_TIME_MS : 0;
+  /** Duration of the dialog close animation. */
+  private _closeAnimationDuration =
+      this._animationsEnabled ? numbers.DIALOG_ANIMATION_CLOSE_TIME_MS : 0;
+  /** Current timer for dialog animations. */
+  private _animationTimer: number|null = null;
+
+  constructor(
+      elementRef: ElementRef, focusTrapFactory: FocusTrapFactory,
+      changeDetectorRef: ChangeDetectorRef, @Optional() @Inject(DOCUMENT) document: any,
+      config: MatDialogConfig, private _ngZone: NgZone,
+      @Optional() @Inject(ANIMATION_MODULE_TYPE) private _animationMode?: string) {
+    super(elementRef, focusTrapFactory, changeDetectorRef, document, config);
+  }
+
+  ngAfterViewInit() {
+    // Delegate to the after view init method from the base dialog container. This
+    // will capture the previously focused element (for later restoring), and will
+    // set focus to the dialog container (assuming that the container is focusable).
+    super.ngAfterViewInit();
+    // Note: Usually we would be able to use the MDC dialog foundation here to handle
+    // the dialog animation for us, but there are a few reasons why we just leverage
+    // their styles and not use the runtime foundation code:
+    //   1. Foundation does not allow us to disable animations.
+    //   2. Foundation contains unnecessary features we don't need and aren't
+    //      tree-shakeable. e.g. background scrim, keyboard event handlers for ESC button.
+    //   3. Foundation uses unnecessary timers for animations in favor of React `setState`
+    //      limitations: https://github.com/material-components/material-components-web/pull/3682.
+    this._startOpenAnimation();
+  }
+
+  ngOnDestroy() {
+    if (this._animationTimer !== null) {
+      clearTimeout(this._animationTimer);
+    }
+  }
+
+  /** Starts the dialog open animation if enabled. */
+  private _startOpenAnimation() {
+    this._animationStateChanged.emit({state: 'opening', totalTime: this._openAnimationDuration});
```

Comment:
It seems like all the places where this is used are stubbing out the interface for Angular's `AnimationEvent`. Since this is an internal event, couldn't we change it to something that is more convenient to pass around?

Reply:
I don't see how this can be made more convenient. From my perspective, this interface is easy to use and understand. It's not too coupled to Angular's `AnimationEvent` since it contains a specific state for the dialog (e.g. `closed`, `closing` and more).

Happy to change to something else if you have any ideas.